### DISCUSSION TITLE
NXT Sensors

### DISCUSSION THREAD

**Discussion by alpha71** [pybricks/support#328] [Q&A]
Hello, i don't know if this topic was already discussed before (honestly i don't find it in the repository!).  I'm fresh in Pybricks and Python as well, so i missed several bases... However... Regarding the use of the NXT sensor i'd like to know how to call them and how to set up the instruction and where can find the _list of the "classes" available_ in each module (for example in nxtdevices or in iodevices...??)?
the module nxtdevices has the classes:
TouchSensor
LightSensor
ColorSensor
UltrasonicSensor
SoundSensor
TemperatureSensor
EnergyMeter

Then how can set the parameters?
for example if i wanna use the Ultrasonic Sensor: 

```python
from pybricks.nxtdevices import UltrasonicSensor

sensor = UltrasonicSensor(port.S1)

dist = sensor.distance()
print (dist)
```

Is it correct?

![immagine](https://user-images.githubusercontent.com/37776195/117456394-6c097f80-af48-11eb-97a1-b3522c179046.png)

**Comment by laurensvalk** [comment]
Yes, that's correct. And you can find all NXT sensor documentation here: https://docs.pybricks.com/en/latest/nxtdevices.html

**Reply by alpha71** [reply]
Hello Laurens, thank you for your feedback i just finished to test all sensor and seems everything fine.
One more question which maybe i can reserve in another discussion topic (if you think!) is if thru the I2C can communicate with ARDUINO (i guess yes for sure....), using it has a slave "advanced" MUX, operating on its registries. Did you already have some try on this matter? I actually did it using EV3 Programming andROBOTC following a work done by Dexter Ind. But i noted how easier is use this communication protocol with pybricks then i guess it's be much easier to manage it even with other devices like Arduino.
